service/utils.go

```diff
@@ -198,3 +199,23 @@ func validateModelName(name string) error {
 
 	return nil
 }
+
+// checkUserPermissions retrieves the user from the JWT.
+// The user will be restricted by the accounts the username can access and their role.
+// If user authentication is turned off, the JWT will irrelevant.
+func checkUserPermissions(w http.ResponseWriter, r *http.Request) (string, error) {
```

Comment:
still thinking for a method name...
getAllowedUser(), getAuthUser()...
checkUserPermissions() sounds to me that you provide the permissions that are checked against something. Here is the token what is checked and from there it is extracted the allowed user (or no user if * is allowed)

I could accept checkUser() even as valid